server/controllers/project.controller/deleteProject.js

```diff
@@ -59,16 +60,25 @@ export default function deleteProject(req, res) {
       return;
     }
 
-    deleteFilesFromS3(project.files);
-
-    project.remove((removeProjectError) => {
-      if (removeProjectError) {
-        sendProjectNotFound();
-        return;
-      }
+    // Wait for the delete operation to complete before proceeding
+    deleteFilesFromS3(project.files)
+      .then(() => {
+        project.remove((removeProjectError) => {
+          if (removeProjectError) {
+            sendProjectNotFound();
```

Comment:
Just want to link this line to issue #3024 to be fixed there -- "not found" isn't the right error.  It should be a 500.